extensions/security/runtime-spi/src/main/java/io/quarkus/security/spi/runtime/OneTimeAuthZCodeAuthenticationRequest.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.security.spi.runtime;
```

Comment:
Where else do you expect it supported apart from Form based authentication ?

Reply:
I don't think we should extend support beyond the form-based authentication because it will never be as good as what proper identity providers (like Keycloak) can do for you. IMO that together with JDBC storage should be enough.

So you ask why it is in the security SPI, that is because Security JPA doesn't know about Vert.x HTTP extension and I'd have to introduce Vert.x HTTP runtime SPI (or we could move this to the Security API, but again, I think we shouldn't encourage others to use this, let's stick to the form-auth and move on).

Reply:
I am open to suggestions.

Reply:
Thanks @michalvavrik, right, if the user  logs in to OIDC then 2FA, etc would be offered there

Reply:
done